src/main/java/seedu/bigpp/parser/Parser.java

```diff
@@ -212,4 +217,27 @@ private Command parseCompareCommand(String arguments) {
                             + "(cpu,gpu,ram,storage,psu,motherboard,cpu-cooler,chassis)");
         }
     }
+
+    private Command parseFilterCommand(String arguments) {
+        String[] inputList = arguments.split(" ", 2);
+        String filterType = inputList[0].toLowerCase();
+        String input = "";
+        if (inputList.length == 2) {
+            input = inputList[1];
+        }
+        switch (filterType) {
+        case "name":
+            return new ViewerFilterNameCommand(input);
+        case "cost":
+            return new ViewerFilterCostCommand(input);
+        case "built":
+            return new ViewerFilterIsBuiltCommand(input);
+        case "clear":
+            return new ViewerFilterClearCommand();
+        default:
+            return new UnrecognizedCommand(
+                    "Invalid component type!, valid types are "
+                            + "(cpu,gpu,ram,storage,psu,motherboard,cpu-cooler,chassis)");
+        }
+    }
```

Comment:
Filter should be one long command, so the filter command will accept multiple flags at once into a long command. You can take a look at how builder list command is being handled.

Reply:
for example

Filter -name <name> -cost /from start /to end -built complete